ui/app/serializers/sync/destination.js

```diff
@@ -6,14 +6,37 @@
 import ApplicationSerializer from 'vault/serializers/application';
 
 export default class SyncDestinationSerializer extends ApplicationSerializer {
-  primaryKey = 'name';
+  // overwrite application serializer's normalizeResponse method
+  normalizeResponse(store, primaryModelClass, payload, id, requestType) {
+    let transformedPayload = payload;
 
-  normalizeItems(payload) {
-    if (payload.data?.connection_details) {
-      const data = { ...payload.data, ...payload.data.connection_details };
-      delete data.connection_details;
-      return data;
+    if (requestType === 'findRecord') {
+      transformedPayload = this._normalizeFindRecord(payload);
     }
-    return super.normalizeItems(...arguments);
+    return super.normalizeResponse(store, primaryModelClass, transformedPayload, id, requestType);
+  }
```

Comment:
I would love to figure out how to still extend the `ApplicationSerializer` while being able to use `normalizeFindRecordResponse` rather than conditionally targeting the `requestType` but this does the trick.

Reply:
Yeah...I couldn't find a way around this. I tried a few things like just interrupt `normalizeItems` and return `payload` but I got a `Maximum call stack exceeded` error 🙃